### ISSUE TITLE
kurento-client (<--kurento-jsonrpc<--utf-8-validate 1.2.x) install fails on node.js v10

### ISSUE THREAD

**Issue by a-sync** [Kurento/bugtracker#260]
**Client libraries**
- Language: Node.js
- Version: **10.4.1** (npm 6.1.0)

**What steps will reproduce the problem?**
1. add kurento-client as dependency (`git://github.com/Kurento/kurento-client-js.git#master`)
2. run `npm install`

**What is the expected result?**
Should build the native sub dependency and install the client.

**What happens instead?**
The issue seems to be the `"utf-8-validate": "1.2.x"` dependency of **kurento-jsonrpc**.  
`nodey-gyp rebuild` fails to build this version of the native file on a node.js v10 runtime.

**Does it happen with one of the tutorials?**
Yes.

**Please provide any additional information below.**
Tried this on windows 7 & ubuntu 16.04 with same results:
  - utf-8-validate 1.2.x install fails with node v10 because of the native build failing (works fine with node v6)
  - utf-8-validate v4 builds are working fine with node v10

I realize this might be a node-gyp / nan / utf-8-validate issue, but since the newer versions work fine, i doubt that there would be any effort from their side to resolve this.

~~Updating the utf-8-validate dependency in kurento-jsonrpc should fix this, but i'm not sure about the side effects of this upgrade.~~  It seems to work fine with utf-8-validate v4.0.2!

Kind regards,
  a-sync

**Comment by haggholm** [comment]
Versions <4 are not supported: https://github.com/websockets/utf-8-validate/issues/75

In order for Kurento to build with newer Node/npm versions, it must upgrade utf-8-validate.

**Comment by georgius1024** [comment]
kurento-jsonrpc dependencies is:
```
  "dependencies": {
    "bufferutil": "1.2.x",
    "inherits": "^2.0.1",
    "utf-8-validate": "1.2.x",
    "ws": "^1.1.1"
  },
```
When i made a fork of kurento-jsonrpc and upgraded utf-8-validate to 5.0.1 it installs ok. But still not working due to changes in validation interface.
So, I'll downgrade to node 8

**Comment by a-sync** [comment]
utf-8-validate interface looks the same in v5 as was in v4
the main difference is the switch to n-api iinm

anyway i also made a fork with a custom kurento-jsonrpc to update the deps and it works fine with node 10  
gihub.com/a-sync/kurento-client-js

**Comment by georgius1024** [comment]
Thank you very much

You fork installs without any error, but wher I run hello world example, scrippt fails 
```
 if (!Validation.isValidUTF8(messageBuffer)) {
                            ^
TypeError: Cannot read property 'isValidUTF8' of undefined
    at ...\node_modules\ws\lib\Receiver.js:473:29
    at ...\node_modules\ws\lib\Receiver.js:361:7
    at ...\node_modules\ws\lib\PerMessageDeflate.js:247:5
    at afterWrite (_stream_writable.js:485:3)
    at onwrite (_stream_writable.js:476:7)
    at InflateRaw.afterTransform (_stream_transform.js:94:3)
    at Zlib.processCallback (zlib.js:629:8)
```
Old version of webscockets or whatever use other validation interface. **sigh**

**Comment by a-sync** [comment]
@georgius1024 check the ws module version in your node_modules folder.
this error seems like you have utf-8-validate version > 2 while the ws version is < 1.1.3.
you might want to try removing the node_modules folder and the package-lock.json and doing a clean install

and to get back on topic: the dependencies and sub dependencies are getting out of date at an alarming rate for the kurento node repos to the point where you can't even run the basic demo apps with a clean install on node 10 :(

**Comment by georgius1024** [comment]
Sadly, but it not works for me. 
1) Use fresh hello-world from https://github.com/Kurento/kurento-tutorial-node
2) Update one line in package.json
```
{
  "name": "kurento-hello-world",
  "version": "6.8.1-dev",
  "private": true,
  "scripts": {
    "postinstall": "cd static && bower install"
  },
  "dependencies": {
    "cookie-parser": "^1.3.5",
    "express": "~4.12.4",
    "express-session": "~1.10.3",
    "minimist": "^1.1.1",
    "ws": "~1.0.1",
    **"kurento-client": "git+https://github.com/a-sync/kurento-client-js"**
  },
  "devDependencies": {
    "bower": "^1.4.1"
  }
}
```
My node version is 11.2.0 and this package installs with a ton of warnings, but no one error.
But when I run example, stil got same error. May be this is a platform issue (Windows)

**Comment by a-sync** [comment]
@georgius1024 i repeated your steps and got the same error 

you also need to update the ws top level dep. to avoid it being used in sub packages

here's a working package.json for the hello-world demo:
```{
  "name": "kurento-hello-world",
  "version": "6.8.1-dev",
  "private": true,
  "scripts": {
    "postinstall": "cd static && bower install"
  },
  "dependencies": {
    "cookie-parser": "^1.3.5",
    "express": "~4.12.4",
    "express-session": "~1.10.3",
    "minimist": "^1.1.1",
    "ws": "1.1.3",
    "kurento-client": "a-sync/kurento-client-js"
  },
  "devDependencies": {
    "bower": "^1.4.1"
  }
}```

this way it works (win10, node 10.13.0)

**Comment by georgius1024** [comment]
It really works for me, Thank you very much!

**Comment by a-sync** [comment]
fixed in v6.11
